at_client/lib/src/service/sync_service_impl.dart

```diff
@@ -460,7 +460,7 @@ class SyncServiceImpl implements SyncService, AtSignChangeListener {
       }
       final serverEncryptedValue = serverCommitEntry['value'];
       final serverMetaData = serverCommitEntry['metadata'];
-      if (serverMetaData != null && serverMetaData[IS_ENCRYPTED] == "true") {
+      if (serverMetaData != null && (serverMetaData[IS_ENCRYPTED] != null && serverMetaData[IS_ENCRYPTED] == "true")) {
```

Comment:
You already had `serverMetaData[IS_ENCRYPTED] == "true"` - by definition, if those values are equal then `serverMetaData[IS_ENCRYPTED] != null` is also true. i.e. this change makes no difference. What problem are you actually trying to solve?

Reply:
If the metadata for the older keys does not have 'IS_ENCRYPTED', then serverMetaData[IS_ENCRYPTED] might result to a null pointer exception. So, having an additional check  to prevent a null pointer